HPCC-17100 Fix esp.xml missing format information

rpastrana opened the pull request:
- Creates xls var for line break, and indent
- Output break and indent before every esp service and binding
- Cleans out extra whitespace in xslt source

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [ ] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

rpastrana commented:
@wangkx Please take a look.

wangkx commented on initfiles/componentfiles/configxml/esp.xsl:
```diff
@@ -181,7 +183,15 @@
+                <xsl:value-of disable-output-escaping="yes" select="$break" />
+                <xsl:value-of disable-output-escaping="yes" select="$indent" />
+                <xsl:value-of disable-output-escaping="yes" select="$indent" />
+                <xsl:value-of disable-output-escaping="yes" select="$indent" />
```
In my option, adding 'break' is good enough. Too many indent lines ...

wangkx replied:
Although esp.xml is in bad format, ECLWatch shows it in a good XML format.

rpastrana replied:
the indent is needed if we're going to format the file content explicitly. The fact that eclwatch formats the xml in the viewer makes me wonder if we should bother w/ this change at all...
@RussWhitehead your thoughts please...

RussWhitehead replied:
IMHO if we can easily format the XML properly then we should.

richardkchapman replied:
I agree

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: f6ede79dc4d5967a412e98ca85e6c14e515f4bdb
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.505 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
Typo "everty" in commit message

rpastrana committed:
HPCC-17100 Fix esp.xml missing format information

- Creates xls var for line break, and indent
- Output break and indent before every esp service and binding

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: c13b97ff7dc19273578172f8d893bd4d4249a348
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.066 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
